README.md

```diff
@@ -0,0 +1,583 @@
+System, Python version (3.7, 3.8, etc.), and Python implementation (CPython, PyPy,
+etc.). For an exact definition, refer to the possible combinations of [PEP 508
+environment markers][environment-markers].
+
+As the resulting `requirements.txt` can differ for each environment, users must
+execute `pip-compile` **on each Python environment separately** to generate a
+`requirements.txt` valid for each said environment. The same `requirements.in` can
+be used as the source file for all environments, using
+[PEP 508 environment markers][environment-markers] as
+needed, the same way it would be done for regular `pip` cross-environment usage.
+
+If the generated `requirements.txt` remains exactly the same for all Python
+environments, then it can be used across Python environments safely. **But** users
+should be careful as any package update can introduce environment-dependent
+dependencies, making any newly generated `requirements.txt` environment-dependent too.
+As a general rule, it's advised that users should still always execute `pip-compile`
+on each targeted Python environment to avoid issues.
+
+### Other useful tools
+
+- [pipdeptree](https://github.com/tox-dev/pipdeptree) to print the dependency tree of the installed packages.
+- `requirements.in`/`requirements.txt` syntax highlighting:
+
+  - [requirements.txt.vim](https://github.com/raimon49/requirements.txt.vim) for Vim.
+  - [Python extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-python.python) for VS Code.
+  - [pip-requirements.el](https://github.com/Wilfred/pip-requirements.el) for Emacs.
+
+### Deprecations
+
+This section lists `pip-tools` features that are currently deprecated.
+
+- In future versions, the `--allow-unsafe` behavior will be enabled by
+  default. Use `--no-allow-unsafe` to keep the old behavior. It is
+  recommended to pass the `--allow-unsafe` now to adapt to the upcoming
+  change.
+- Legacy resolver is deprecated and will be removed in future versions.
+  Use `--resolver=backtracking` instead.
+
+### A Note on Resolvers
+
+You can choose from either the legacy or the backtracking resolver.
+The backtracking resolver is recommended, and will become the default
+with the 7.0 release.
+
+Use it now with the `--resolver=backtracking` option to `pip-compile`.
+
+The legacy resolver will occasionally fail to resolve dependencies. The
+backtracking resolver is more robust, but can take longer to run in
+general.
+
+You can continue using the legacy resolver with `--resolver=legacy`.
+
+### Versions and compatibility
+
+The table below summarizes the latest `pip-tools` versions with the required
+`pip` and Python versions. Generally, `pip-tools` supports the same Python
+versions as the required `pip` versions.
+
+| pip-tools      | pip            | Python         |
+| -------------- | -------------- | -------------- |
+| 4.5.\*         | 8.1.3 - 20.0.2 | 2.7, 3.5 - 3.8 |
```

Comment:
Could the `*` be breaking this table?

Reply:
But it's escaped?

Reply:
Yes, though I'm not sure Markdown has a concept of escaping like RST. So it might be parser-dependent. GH uses GFM and Sphinx has MyST integrated. This is why more experimentation may be needed to make sure it works universally.